shadowsocks/tcp.go

```diff
@@ -210,9 +213,12 @@ func (s *tcpService) Start() {
 			timeToCipher = time.Now().Sub(findStartTime)
 
 			if err != nil {
+				logger.Debugf("Failed to find a valid cipher after reading %v bytes: %v", proxyMetrics.ClientProxy, err)
+				io.Copy(ioutil.Discard, clientConn) // drain socket
 				return onet.NewConnectionError("ERR_CIPHER", "Failed to find a valid cipher", err)
 			}
 
+			clientConn.SetReadDeadline(time.Time{})
```

Comment:
Comment: "Clear the authentication deadline"

Reply:
done